dashboard/test/models/pd/workshop_test.rb

```diff
@@ -697,4 +736,11 @@ def session_on(day_offset, start_offset, end_offset)
   def today
     Date.today.in_time_zone
   end
+
+  def setup_mock_time_constraint(constraint_hash)
+    Pd::Workshop.send(:remove_const, :TIME_CONSTRAINTS_BY_SUBJECT)
+    mock_time_constraints_by_subject = mock
+    mock_time_constraints_by_subject.expects(:[]).returns(constraint_hash)
+    Pd::Workshop.const_set :TIME_CONSTRAINTS_BY_SUBJECT, mock_time_constraints_by_subject
+  end
```

Comment:
D'oh of course. Overriding this constant is bleeding state into other tests and causing them to fail. I don't think the tests that use this add much value, since those cases are already covered in the other tests, so I'm going to remove them. 

Other options are wrapping the const in a class method which can be stubbed, or saving the original value here and re-setting it in teardown.